Libraries/Azure.psm1

```diff
@@ -774,55 +774,58 @@ Function Get-AllDeploymentData($ResourceGroups)
 
         foreach ($testVM in $RGVMs)
         {
+            $testVMDetails = Get-AzVM -ResourceGroupName $ResourceGroup -Name $testVM.Name
```

Comment:
Put this command above for-each loop `$AllVMs = Get-AzVM -ResourceGroupName $ResourceGroup`
And in for loop, `$testVMDetails = $AllVMs | Where { $_.Name -eq $testVM.Name }` This will save a lot of time.